experimental/dds/tree2/src/feature-libraries/sequence-field/utils.ts

```diff
@@ -369,8 +369,12 @@ export function tryExtendMark<T>(lhs: Mark<T>, rhs: Readonly<Mark<T>>): boolean
 	}
 	const type = rhs.type;
 	if (type === NoopMarkType) {
-		(lhs as NoopMark).count += rhs.count;
-		return true;
+		if (areEqualCellIds(getCellId(lhs, undefined), getCellId(rhs, undefined))) {
+			(lhs as NoopMark).count += rhs.count;
+			return true;
+		} else {
+			return false;
+		}
```

Comment:
This function returns false by default at the end, so I think it'd be fine to get rid of the else block and combine the two if statements.

Reply:
I made this change but had to add a `type !== NoopMarkType` to a few more blocks below.